### ISSUE TITLE
Remove OFP frontend

### ISSUE THREAD

**Issue by mlange05** [ecmwf-ifs/loki#209]
The OFP frontend is getting increasingly hard to maintain. A recent brush with some of its inconsistencies (mostly due to skew in the generated AST node scoping due to upstream bugs) has highlighted that in particular better handling of functions/procedures will become very painful if we are to maintain this. 

I've checked and the most recent release (0.6.1) is not compatible with our frontend implementation, but it does still have many of the scoping quirks that we tend to bend over backwards to circumvent. As we have little use for it that we cannot deal with via other frontends, I propose we now fully deprecate and/or entirely remove the OFP frontend. This should improve test throughput and remove some of the special-casing, ideally before we continue to align procedures as proper IR nodes.

I'm happy to implement and test this, but input and feedback are very welcome (@reuterbal ).

**Comment by reuterbal** [comment]
I agree. OFP is dead slow, which, among all the other issues, would make it undesirable for any productive use anyway. There is value in maintaining more than one frontend but arguably we should rather spend time on exploring other alternatives than fixing up OFP.

**Comment by reuterbal** [comment]
Following the ideas tossed around in #406 and the thin thread we have thrown as a lifeline to OMNI in #408, the proposal is to proceed to deprecating OFP:

- [ ] Add a big warning in every outward API of OFP that makes clear that OFP will be removed in the next release and is no longer maintained and tested
- [ ] Remove OFP from the list returned by `available_frontends()`
- [ ] Clean up the test base to remove every trace of OFP from there
- [ ] After the next Loki release, remove OFP and its traces from the remaining Loki code base
